ci/scripts/build.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+set -e
+
+# This sets $ARCH and $PLATFORM env variables
+source "$(dirname "$0")/../../scripts/get_arch_and_platform.sh"
+
+if [[ $PLATFORM == "mac" ]]; then
+   BUILD_SCRIPT="build-mac.sh"
+elif [[ $PLATFORM == "linux" ]]; then
+   BUILD_SCRIPT="build-linux.sh"
+else
+   echo "Invalid target use one of [mac, linux]"
+   exit 1
+fi
+
+function setup_environment() {
+  export BRANCH=${BRANCH_NAME:-dev}
+  export POSTGRES_USER=postgres
+  export DEBIAN_FRONTEND=noninteractive
+  export PG_VERSION=${PG_VERSION:-15}
+  export GITHUB_OUTPUT=${GITHUB_OUTPUT:-/dev/null}
+}
+
+function clone_or_use_source() {
+  if [ -z ${USE_SOURCE} ]; then
+    # Clone from git
+    cd /tmp
+    git clone --recursive https://github.com/lanterndata/lantern.git -b $BRANCH
+  else 
+    # Use already checkouted code
+    shopt -s dotglob
+    rm -rf /tmp/lantern
+    mkdir -p /tmp/lantern
+    cp -r -P ./* /tmp/lantern/
+  fi
+}
+
+function build_and_install() {
+  cd /tmp/lantern
+  mkdir build
+  cd build
+
+  flags="-DUSEARCH_NO_MARCH_NATIVE=ON"
+  
+  # On macs -static-libstdc++ throws warning
```

Comment:
Let's actually modify the cmake file to not pass this flag in mac build.
Then we will have one less warning and will get rid of the `if` below